### PR TITLE
Add systemd for RHEL & Ubuntu

### PR DESCRIPTION
**Pull request by matthew-watson1**

### CODE REVIEW

File: tasks/install.yml [hunk 1]
```diff
@@ -46,12 +46,21 @@
...
     state: link
 
-- name: Install Consul service
+- name: Install Systemd Consul service
+  become: yes
+  template:
+    dest: /etc/systemd/system/consul.service
+    mode: 0755
+    src: systemd/systemd.j2
+  when: ansible_service_mgr == 'systemd'
```
Comment (lobsterdore) [inline tasks/install.yml:55]: 👍

File: tests/vagrant/Vagrantfile [hunk 2]
```diff
@@ -6,15 +6,15 @@ VAGRANTFILE_API_VERSION = "2"
 
 Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
 
-	config.vm.box = "ubuntu/trusty64"
+	config.vm.box = "ubuntu/xenial64"
```
Comment (lobsterdore) [inline tests/vagrant/Vagrantfile]: Can you leave this as trusty64 for now?

File: tests/vagrant/Vagrantfile [hunk 3]
```diff
@@ -6,15 +6,15 @@ VAGRANTFILE_API_VERSION = "2"
...
 	config.vm.box_check_update = false
 
 	config.ssh.forward_agent = true
+	config.vm.network :private_network, ip: '192.168.10.10'
```
Comment (lobsterdore) [inline tests/vagrant/Vagrantfile]: Why add a private IP address if no hostname or DNS is being setup?

File: templates/defaults.j2 [hunk 4]
```diff
@@ -1,5 +1,6 @@
 CONSUL_STARTOPS="{{ consul.startopts }}"
 BIND_INTERFACE="{{ consul.bind_interface | default('eth0') }}"
+BIND="$(ifconfig {{  consul.bind_interface | default('eth0') }} | grep "inet addr" | awk '{ print substr($2,6) }')"
```
Comment (lobsterdore) [inline templates/defaults.j2]: Can you reuse BIND_INTERFACE and update the upstart config appropriately? I think we should keep the two init methods aligned as much as possible.

File: tests/vagrant/Vagrantfile (hunk 2 above)
Comment (matthew-watson1) [reply tests/vagrant/Vagrantfile]: Sure, will do

File: tests/vagrant/Vagrantfile (hunk 3 above)
Comment (matthew-watson1) [reply tests/vagrant/Vagrantfile]: I keep encountering the following error without it:

NFS requires a host-only network to be created.
Please add a host-only network to the machine (with either DHCP or a
static IP) for NFS to work.

Comment (lobsterdore) [reply tests/vagrant/Vagrantfile]: Hmmmm you must have something that is trying to share a folder via NFS, have you got anything in ~/.vagrant.d/Vagrantfile ? Either that or for some reason Vagrant is trying to mount the default share via NFS, which it shouldn't do.

Comment (matthew-watson1) [reply tests/vagrant/Vagrantfile]: Ah, It was the vagrant-cachier plugin that was breaking it. Thanks!

**Comment by matthew-watson1** [comment]
@lobsterdore Looks okay now?

**Comment by lobsterdore** [comment]
Yep looks good, the issue ID in the commit message should be 6 as opposed to 5 though, the actual ID of the issue you created is 6. Can you update the PR title as well to match the commit message? You will need to get another approval as well, such as @racer1680, @chroche or @wojtek-oledzki  .

**Comment by matthew-watson1** [comment]
@lobsterdore , @racer1680, @chroche or @wojtek-oledzki. I am closing this pull request and opening a new one with updated details #8

**Pull request closed without merging**